docs/lib/introduction.md

```diff
@@ -0,0 +1,789 @@
+[`ocireg`](../../pkg/contexts/ocm/repositories/ocireg) package and the [`ctf`](../../pkg/contexts/ocm/repositories/ctf)
+merely provide the functionality to convert, or rather wrap, their _OCI Repository Spec_ to an _OCM Repository Spec_
+(remember, spec types were pretty much serializable representations and factories for the actual type, as explained
+[here](introduction/context/registry/registry.go)).  
+For each _OCI Repository Type_, there also exists a corresponding _OCM Repository Type_.
+Thus, as there is an **oci** _repository type_ `OCIRegistry`, there is also an **ocm** _repository type_ `OCIRegistry`,
+as there is an **oci** _repository type_ `CommonTransportFormat`, there is also an **ocm** _repository type_
+`CommonTransportFormat`.
+Within the ocm-lib this is implemented through an **addition to the factory-based type registry** concept. In case the
+does not find an entry for a certain type (within the Decode method), this
+[implementation (defaultScheme)](../../pkg/runtime/scheme.go) of a factory-based type registry has a _fallback factory_
+(called `defaultdecoder`). For the OCM Repository Type Registry, this _fallback factory_ is usually
+[initialized](../../pkg/contexts/ocm/builder.go) with a Type Registry itself, the
+[`DefaultDelegationRegistry`](../../pkg/contexts/ocm/repositories/genericocireg/type.go) which is based on the
+[`delegationRegistry`](../../pkg/contexts/ocm/internal/delegation.go). This Type Registry's iterates through all its
+factories and checks whether they knew that type. Currently, the only factory known by this _fallback Type Registry_ is
+the one registered in the [`genericocireg`](../../pkg/contexts/ocm/repositories/genericocireg/type.go) package which
+uses the **oci** _repository types registry_ to perform the unmarshaling (thus, if I registered the type in the
+oci repository types registry, it is automatically available for the ocm repository registry).
+
+**Other Repository Types:**
+Besides _OCM_ and _OCI Repository Types_, there currently also exist a number of
+[_Credential Repository Types_](../../pkg/contexts/credentials/repositories) that have to implement the following
+[interface](../../pkg/contexts/credentials/internal/repository.go):
+
+```
+type Repository interface {
+   ExistsCredentials(name string) (bool, error)
+   LookupCredentials(name string) (Credentials, error)
+   WriteCredentials(name string, creds Credentials) (Credentials, error)
+}
+```
+
+### Section 3.2.2 - Access Type Scheme
+
+An `AccessTypeScheme` is an object that maps a number of _access types (= model type)_ to corresponding
+_factories_ providing an access spec object which is again a _factory_ for providing an object of a specific type or
+with
+a specific configuration for dealing with this _access type (= model type)_.
+
+From the perspective of the ocm-lib, any _go type_ that implements the following
+[interface](../../pkg/contexts/ocm/internal/accesstypes.go) can be considered an _OCI Repository_:
```

Comment:
Why can the existence of a go type implementing this AccessMethod interface be considered a OCI repository ? Below, an implementation for S3 or GitHub are mentioned as examples, which is not an OCI repositories, so maybe rather "OCM Repository" is meant here ?

Reply:
That's a copy-paste mistake. It should be ...can be considered an _Access Method_.